### PR TITLE
[WIP] Update exception classes

### PR DESCRIPTION
**Pull request by MridulS**
@jfinkels

### CODE REVIEW

**Comment by MridulS** [comment]
Instead of merging one huge PR, I guess we can start updating errors like we do for PEP8 changes. Update them only when you work on the file.

**Comment by dschult** [comment]
Is this still a [WIP] ?  

Also, you should update the docs at doc/source/reference/exceptions.rst

**Comment by jfinkels** [comment]
Well updating an exception name in one file has changes throughout the code, wherever the exception is raised. This is different from making local style changes in a single file.

I don't think it really matters how the changes are made, but if these changes are definitely going to be made, they must be totally finished within one release cycle (say networkx-2.0), not spread across multiple releases (i.e. half the exceptions renamed by networkx-2.0, the other half by networkx-2.1).

File: networkx/exception.py [hunk 1]
```diff
@@ -52,3 +52,9 @@ class NetworkXUnbounded(NetworkXAlgorithmError):
 
 class NetworkXNotImplemented(NetworkXException):
     """Exception raised by algorithms not implemented for a type of graph."""
+
+class NotFound(NetworkXException):
+    """Exception"""
```
Comment (jfinkels) [inline networkx/exception.py]: This is not a good docstring.

**Comment by jfinkels** [comment]
As for the exception itself, I think this is the right idea.

**Comment by hagberg** [comment]
Needs updated docstring and merge.

**Pull request merged**